test/over_react_redux/hooks/use_selector_test.dart

```diff
@@ -0,0 +1,169 @@
+
+          expectCountValue(jacket, 0);
+          await clickSiblingConnectedIncrementButton(jacket);
+          expectCountValue(jacket, 1);
+        });
+
+        test('unless a custom equalityFn returns true', () async {
+          jacket = mount(
+            (ReduxProvider()..store = counterStore)(
+              (CounterFn()..countEqualityFn = (nextCount, prevCount) {
+                return nextCount == 1;
+              })(),
+              // Use a sibling connected component for dispatching actions in these tests
+              // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+              (ConnectedCounter()..addTestId('sibling-counter'))(),
+            ), attachedToDocument: true);
+
+          expectCountValue(jacket, 0);
+          await clickSiblingConnectedIncrementButton(jacket);
+          expectCountValue(jacket, 0);
+        });
+      });
+
+      // Exercise js interop wrapping / unwrapping of dart types
+      group('a selector with a Dart value is updated', () {
+        test('', () async {
+          jacket = mount(
+            (ReduxProvider()..store = counterStore)(
+              ModelCounterFn()(),
+              // Use a sibling connected component for dispatching actions in these tests
+              // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+              (ConnectedCounter()..addTestId('sibling-counter'))(),
+            ), attachedToDocument: true);
+
+          expectModelCountValue(jacket, 0);
+          await clickSiblingConnectedModelCountIncrementButton(jacket);
+          expectModelCountValue(jacket, 1);
+        });
+
+        test('unless a custom equalityFn returns true', () async {
+          jacket = mount(
+            (ReduxProvider()..store = counterStore)(
+              (ModelCounterFn()..modelCountEqualityFn = (nextModel, prevModel) {
+                return nextModel.count == 1;
+              })(),
+              // Use a sibling connected component for dispatching actions in these tests
+              // that shouldn't rely on `useDispatch` to ensure the subscription to context is wired up correctly
+              (ConnectedCounter()..addTestId('sibling-counter'))(),
+            ), attachedToDocument: true);
+
+          expectModelCountValue(jacket, 0);
+          await clickSiblingConnectedModelCountIncrementButton(jacket);
+          expectModelCountValue(jacket, 0);
+        });
+      });
+    });
+
+    group('subscribes to custom context when createSelectorHook is used and redraws when', () {
+      group('the selector value is updated', () {
+        test('', () async {
+          jacket = mount(
+              (ReduxProvider()..store = counterStore)(
+                (ReduxProvider()
+                  ..store = bigCounterStore
+                  ..context = bigCounterContext
+                )(
+                  CustomContextCounterFn()(),
```

Comment:
I will say that using shared component code, in another file, makes it harder to tell at a glance how exactly these things are being tested, since you have to dig to find the `createSelectorHook` call as opposed to it being in this test/group block